Add ClickHouse managed credentials examples (#9)
Aiven-Open/terraform-example-projects: clickhouse/clickhouse_integrations/credentials/README.md

# ClickHouse managed credentials

> This example creates a ClickHouse cluster and integrates it with credentials stored as service integration endpoints.

Reviewer: Would this be accurate?
```suggestion
This example creates a ClickHouse cluster and integrates it with credentials for other services stored as service integration endpoints.
```

Author: The credentials are mostly meant to be useful for external endpoints (like S3 and databases outside Aiven). Restricting the scope to "credentials for other services" (implicitly, Aiven ones) may bring confusion: for instance, the sentence right underneath talks about S3, which wouldn't qualify as "other service".

Reviewer: Makes sense! Thanks for the explanation. :)

> An S3 bucket is defined by Terraform variables and creates an service integration endpoint resource. 

Reviewer: ```suggestion
The integration endpoint with your S3 bucket is defined by the S3 URL and access keys stored in Terraform variables. 
```

> The endpoint is made available to the ClickHouse cluster through a managed credentials integration.

Reviewer: We could link to the conceptual info in the docs here.
```suggestion
The endpoint is made available to the ClickHouse cluster through a [managed credentials integration](https://aiven.io/docs/products/clickhouse/concepts/data-integration-overview#managed-credentials-integration).
```

> Example PostgreSQL, MySQL and ClickHouse services are created to show how remote databases would be integrated. 

Reviewer: ```suggestion
Aiven for PostgreSQL, MySQL, and ClickHouse services are created to show how remote databases would be integrated. 
```

> Managed credentials allow ClickHouse users to access them with the PostgreSQL & MySQL table engines, and the `remoteSecure` function.

Reviewer: ```suggestion
Managed credentials allow ClickHouse users to access them with the PostgreSQL and MySQL table engines, and the `remoteSecure` function.
```

> ## Prerequisites

Author: Agreed, thank you

* [Install Terraform](https://www.terraform.io/downloads)
* [Sign up for Aiven](https://console.aiven.io/signup?utm_source=github&utm_medium=organic&utm_campaign=devportal&utm_content=repo)
* [Create an authentication token](https://docs.aiven.io/docs/platform/howto/create_authentication_token.html)

> ## Create the example resources

Reviewer: This section, too, can be removed. It's only on the main page and added to the get started and org setup examples (since they are starting points).

Author: Removed!

    $ terraform --version 
    ```

    The output is similar to the following:

    ```sh
    Terraform version: 1.9.5
    + provider registry.terraform.io/aiven/aiven v4.9.2
    ```

2. Clone this repository.

3. Replace the placeholders in the `get-started.tf` file. It's recommended to use your organization name as a prefix for the project name.

4. Initialize Terraform:

```sh
$ terraform init
```

The output is similar to the following:

```sh

Initializing the backend...

Initializing provider plugins...
- Finding aiven/aiven versions matching ">= 4.0.0, < 5.0.0"...
- Installing aiven/aiven v4.25.0...
- Installed aiven/aiven v4.25.0
...
Terraform has been successfully initialized!
...
```

5. To create an execution plan and preview the changes that will be made, run:

```sh
$ terraform plan

```

6. To deploy your changes, run the following and enter yes to confirm:

```sh
$ terraform apply 
```

## Verify the changes in the Aiven Console

You can see the services, integration endpoints and managed credentials integrations in the [Aiven Console](https://console.aiven.io/):

1. In the project services list, you can see the ClickHouse, PostgreSQL, and MySQL services.

> 2. In the Integration endpoints list, you can see the S3 endpoint and the external ClickHouse, PostgreSQL, and MySQL endpoints.

Reviewer: ```suggestion
2. Click **Integration endpoints** to see the S3, external ClickHouse, PostgreSQL, and MySQL endpoints.
```

> 3. In the overview page for the `clickhouse-gcp-eu` service, the managed credentials are show in the Data Integrations section and the Data pipeline view shows the four integrations as active.

Reviewer: Small rewrite to really guide the user to the correct section. I'm not super familiar with ClickHouse or finding service info in the UI in general, so double-check I didn't get something wrong!

```suggestion
3. Click **Services** and select the `clickhouse-gcp-eu` service.
4. To view the managed credentials go to the **Integrations** section.
5. To see the four integrations, go to the **Data pipeline** section.
```

Author: That is indeed clearer, thanks.

> ## Clean up

Reviewer: This section is also included in the general instructions.

Author: Removed
